generate_schedule.py

```diff
@@ -105,33 +105,34 @@ def clamp_to_gc(power):
 
         # basic value: external load, feed-in power
         base_flex = sum([gc.get_current_load() for gc in s.world_state.grid_connectors.values()])
-
         num_cars_present = 0
-
         # update vehicles
         for vid, v in s.world_state.vehicles.items():
             if v.connected_charging_station is None:
                 # vehicle not present: reset info, add to power needed in last interval
                 power_needed += cars[vid][1]
                 cars[vid] = [0, 0, 0]
             else:
-                num_cars_present += 1
-                if cars[vid][0] == 0:
-                    # just arrived
-                    charging_power = v.battery.loading_curve.max_power
-                    delta_soc = max(v.get_delta_soc(), 0)
-                    # scale with remaining steps
-                    if v.estimated_time_of_departure is not None:
-                        dep = v.estimated_time_of_departure
-                        # try to understand this one
-                        dep = -((scenario.start_time - dep) // s.interval)
-                        factor = min((scenario.n_intervals - step_i) / (dep - step_i), 1)
-                        delta_soc *= factor
-                    vehicle_energy_needed = (delta_soc * v.battery.capacity) / v.battery.efficiency
-                    v.battery.soc = max(v.battery.soc, v.desired_soc)
-                    v2g = (v.battery.get_available_power(s.interval)
-                           * v.vehicle_type.v2g_power_factor) if v.vehicle_type.v2g else 0
-                    cars[vid] = [charging_power, vehicle_energy_needed, v2g]
+                if s.world_state.charging_stations[v.connected_charging_station].parent == \
```

Comment:
compare cs.parent with gcID. For better readability, I would declare the current cs at the beginning of the vehicle-loop:
```
for vid, v in ...:
  cs_id = v.connected_charging_station
  if cs_id is None:
    ...
  else:
    # vehicle at charging station: charge full, keep track of needed and available energy
    cs = s.world_state.charging...
    if cs.parent == gcID:
      ...
```